lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -33,6 +33,34 @@ static int getTensorRank(Value tensor) {
   return tensorRank;
 }
 
+namespace {
+class DecomposeAtenSelectIntOp : public OpRewritePattern<AtenSelectIntOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenSelectIntOp op, PatternRewriter &rewriter) const override{
+    Location loc = op.getLoc();
+    Value one = rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value end = rewriter.create<AtenAddIntOp>(loc, one.getType(), op.index(), one);
+    Value self = op.self();
+    BaseTensorType tensorType = self.getType().cast<BaseTensorType>();
+    
+    ArrayRef<int64_t> sizes = tensorType.getSizes();
+    SmallVector<int64_t> resultSizes;
+    int64_t dim, rank;
+    rank = sizes.size();
+    if (!matchPattern(op.dim(),  m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "SelectInt requires dim be constant int.");
+    for (int64_t i=0; i<rank; i++){
+      if (!(i==dim))
```

Comment:
nit: please clang-format this pattern. And for this line `i != dim` is more readable.